datadome/resource_custom_rule.go

```diff
@@ -2,42 +2,72 @@ package datadome
 
 import (
 	"context"
+	"fmt"
 	"strconv"
+	"time"
 
 	dd "github.com/datadome/terraform-provider/datadome-client-go"
+	"github.com/hashicorp/go-cty/cty"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+// resourceCustomRule define the CRUD operations and the schema definition for DataDome custom rules.
 func resourceCustomRule() *schema.Resource {
 	return &schema.Resource{
 		CreateContext: resourceCustomRuleCreate,
 		ReadContext:   resourceCustomRuleRead,
 		UpdateContext: resourceCustomRuleUpdate,
 		DeleteContext: resourceCustomRuleDelete,
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
-				ForceNew: true,
 			},
-			"query": &schema.Schema{
+			"query": {
 				Type:     schema.TypeString,
 				Required: true,
 			},
-			"response": &schema.Schema{
+			"response": {
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateDiagFunc: func(v any, p cty.Path) diag.Diagnostics {
+					var diags diag.Diagnostics
+					value := v.(string)
+					if !(value == "whitelist" || value == "captcha" || value == "block") {
+						diag := diag.Diagnostic{
+							Severity: diag.Error,
+							Summary:  "wrong value",
+							Detail:   fmt.Sprintf("%q is not an acceptable response", value),
+						}
+						diags = append(diags, diag)
+					}
+					return diags
+				},
 			},
-			"priority": &schema.Schema{
+			"priority": {
 				Type:     schema.TypeString,
 				Optional: true,
+				ValidateDiagFunc: func(v any, p cty.Path) diag.Diagnostics {
+					var diags diag.Diagnostics
+					value := v.(string)
+					if !(value == "high" || value == "normal" || value == "low") {
+						diag := diag.Diagnostic{
+							Severity: diag.Error,
+							Summary:  "wrong value",
+							Detail:   fmt.Sprintf("%q is not an acceptable priority", value),
+						}
+						diags = append(diags, diag)
+					}
+					return diags
+				},
+				Default: "high",
 			},
-			"endpoint_type": &schema.Schema{
+			"endpoint_type": {
```

Comment:
This seems to be a restricted list too:
- web
- api-app-mobile
- rss
- api
- login
- submit

Reply:
Changes applied in https://github.com/DataDome/terraform-provider-datadome/pull/24/commits/966615d2312031016acf1cdb66d9115b5a4c33c3

Add the following:
- `ValidateDiagFunc` to restrict to the enum you mentioned
- `Default` value to `web` as it is specified in the API Specification